pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -1106,12 +1109,22 @@ func (c InstallerController) ensureRequiredResourcesExist(ctx context.Context, r
 }
 
 func (c InstallerController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	operatorSpec, originalOperatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	operatorSpec, originalOperatorStatus, operatorResourceVersion, err := c.operatorClient.GetStaticPodOperatorState()
 	if err != nil {
 		return err
 	}
 	operatorStatus := originalOperatorStatus.DeepCopy()
 
+	operatorRV, err := strconv.ParseUint(operatorResourceVersion, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	if operatorRV < c.lastAppliedRV {
+		klog.V(4).Info("Skipping installer controller sync, StaticPodOperator lister is out of date")
```

Comment:
Nit: Out of date might be OK, but to be precise this should say that it may not have observed the effect of its most recent write.